numpy/core/tests/test_multiarray.py

```diff
@@ -1073,34 +1073,60 @@ def test_array_too_big(self):
             assert_raises(ValueError, np.ndarray, buffer=buf, strides=(0,),
                           shape=(max_bytes//itemsize + 1,), dtype=dtype)
 
-    def test_jagged_ndim_object(self):
+    def _ragged_creation(self, seq):
+        # without dtype=object, the ragged object should raise
+        with assert_warns(DeprecationWarning):
+            a = np.array(seq)
+        b = np.array(seq, dtype=object)
+        assert_equal(a, b)
+        return b
```

Comment:
Since this is a C deprecation, you should put this test in `test_deprecated`, where it has access to `assert_deprecated` which tests for both warnings and exceptions

Reply:
some of this is testing the deprecation, but the original tests checked functionality. Should I duplicate the new tests of the current functionality, or move the whole lot of them, old and new, to `test_deprecations`?

Reply:
Ah, I hadn't noticed this is a helper function not a test.

I think a single test in `test_deprecation` would suffice, possibly with a comment pointing to this function as something that needs changing when the deprecation is removed.

Reply:
added